segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/S3ClientWrapper.java

```diff
@@ -0,0 +1,279 @@
+    public void putObject(String bucketName, String key, Range range, Object content) {
+
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        try (FileChannel channel = FileChannel.open(path, StandardOpenOption.WRITE)) {
+
+            long startOffset = range.getFirst();
+            long length = range.getLast() + 1 - range.getFirst();
+            do {
+                long bytesTransferred = channel.transferFrom(Channels.newChannel((InputStream) content),
+                        range.getFirst(), range.getLast() + 1 - range.getFirst());
+                length -= bytesTransferred;
+                startOffset += bytesTransferred;
+            } while (length > 0);
+
+            AclSize aclKey = aclMap.getAclSizeForObject(key);
+            aclMap.updateObject(key, aclKey.withSize(range.getLast() + 1));
+        } catch (IOException e) {
+            throw new BadOffsetException(key, 0, 0);
+        }
+    }
+
+    @Synchronized
+    @Override
+    public void setObjectAcl(String bucketName, String key, AccessControlList acl) {
+        AclSize retVal = aclMap.getAclSizeForObject(key);
+        if (retVal == null) {
+            throw new S3Exception("NoObject", 404, "NoSuchKey", key);
+        }
+        aclMap.updateObject(key, retVal.withAcl(acl));
+    }
+
+    @Synchronized
+    @Override
+    public void setObjectAcl(SetObjectAclRequest request) {
+        AclSize retVal = aclMap.getAclSizeForObject(request.getKey());
+        if (retVal == null) {
+            throw new S3Exception("NoObject", 404, "NoSuchKey", request.getKey());
+        }
+        aclMap.updateObject(request.getKey(), retVal.withAcl(request.getAcl()));
+    }
+
+    @Synchronized
+    @Override
+    public AccessControlList getObjectAcl(String bucketName, String key) {
+        AclSize retVal = aclMap.getAclSizeForObject(key);
+        if (retVal == null) {
+            throw new S3Exception("NoObject", 404, "NoSuchKey", key);
+        }
+        return retVal.getAcl();
+    }
+
+    @Override
+    public void deleteObject(String bucketName, String key) {
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        try {
+            Files.delete(path);
+        } catch (IOException e) {
+            throw new S3Exception("NoSuchKey", 404, "NoSuchKey", "");
+        }
+        aclMap.deleteObject(key);
+    }
+
+    @Override
+    public ListObjectsResult listObjects(String bucketName, String prefix) {
+        ListObjectsResult result = new ListObjectsResult();
+        ArrayList<S3Object> list = new ArrayList<>();
+        Path path = Paths.get(this.baseDir, bucketName, prefix);
+        try {
+            if (Files.exists(path)) {
+                if (Files.isDirectory(path)) {
+                    Files.list(path).forEach((file) -> {
+                        S3Object object = new S3Object();
+                        object.setKey(file.toString().replaceFirst(Paths.get(this.baseDir, bucketName).toString(), ""));
```

Comment:
This and line 175 do the same thing. Can you extract this into a method which explains what it's doing and use it in both places?